src/Microsoft.AspNet.OData/Formatter/ODataMediaTypeFormatter.cs

```diff
@@ -291,6 +291,21 @@ public override Task<object> ReadFromStreamAsync(Type type, Stream readStream, H
 
             try
             {
+                if (typeof(Stream).IsAssignableFrom(type))
+                {
+                    // Ideally, it should go into the "ODataRawValueSerializer",
+                    // However, OData lib doesn't provide the method to overwrite/copyto stream
+                    // So, Here's the workaround
+                    Stream objStream = value as Stream;
+                    if (objStream != null)
+                    {
+                        objStream.CopyToAsync(writeStream);
```

Comment:
Do we need to await asynchronous operations to ensure its completion?
```csharp
                    if (objStream != null)
                    {
                        await objStream.CopyToAsync(writeStream);
                        await writeStream.FlushAsync();
                    }
```

Reply:
No, we don't.
And C# compile doesn't allow to use await in a non-async method.

Reply:
Yes, but if we don't wait for CopyToAsync to complete, then FlushAsync could interrupt the CopyTo.  So I think we should either mark this method as async and do the awaits (which is the right thing once we make ODL fully async), or we should call the synchronous objectStream.CopyTo(writeStream).

---
In reply to: [566634550](https://github.com/OData/WebApi/pull/2401#discussion_r566634550) [](ancestors = 566634550)

Reply:
`Task WriteResponseBodyAsync` is a vritual method, we can't change the syntax of this method.

I updated the codes using await. please take a look again.